class UserError(Exception):
    pass;

class InternalError(Exception):
    pass;

class EvaluationError(Exception):
    pass;
